core/src/test/java/com/coderanch/blackjack/HandTest.java

```diff
@@ -0,0 +1,200 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.coderanch.blackjack;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import com.coderanch.blackjack.Card.Rank;
+import com.coderanch.blackjack.Card.Suit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import static org.junit.Assert.assertThrows;
+import org.junit.experimental.theories.DataPoints;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
+
+
+/**
+ * Tests the {@link Hand} class.
+ */
+@RunWith(Theories.class)
+public final class HandTest {
+
+    /**
+     * Hands to test scoring.
+     */
+    @DataPoints
+    public static final List<HandTestArgument> HANDS =
+        List.of(
+            new HandTestArgument(
+                List.of(new Card(Rank.ACE, Suit.CLUBS),
+                    new Card(Rank.ACE, Suit.CLUBS),
+                    new Card(Rank.KING, Suit.CLUBS),
+                    new Card(Rank.EIGHT, Suit.CLUBS),
+                    new Card(Rank.ACE, Suit.CLUBS)),
+                21
+            ),
+
+            new HandTestArgument(
+                List.of(new Card(Rank.QUEEN, Suit.CLUBS),
+                    new Card(Rank.EIGHT, Suit.CLUBS)),
+                18
+            ),
+
+            new HandTestArgument(
+                List.of(new Card(Rank.QUEEN, Suit.CLUBS),
+                    new Card(Rank.ACE, Suit.CLUBS)),
+                21
+            ),
+
+            new HandTestArgument(
+                List.of(new Card(Rank.KING, Suit.CLUBS),
+                    new Card(Rank.KING, Suit.CLUBS),
+                    new Card(Rank.KING, Suit.CLUBS),
+                    new Card(Rank.KING, Suit.CLUBS),
+                    new Card(Rank.KING, Suit.CLUBS)),
+                50
+            ),
+
+            new HandTestArgument(
+                List.of(new Card(Rank.ACE, Suit.CLUBS),
+                    new Card(Rank.ACE, Suit.CLUBS),
+                    new Card(Rank.ACE, Suit.CLUBS),
+                    new Card(Rank.KING, Suit.CLUBS),
+                    new Card(Rank.KING, Suit.CLUBS)),
+                23
+            ),
+
+            new HandTestArgument(
+                List.of(new Card(Rank.ACE, Suit.CLUBS),
+                    new Card(Rank.ACE, Suit.CLUBS),
+                    new Card(Rank.NINE, Suit.CLUBS)),
+                21
+            )
+        );
+
+    /**
+     * Cards to test.
+     */
+    @DataPoints
+    public static final Set<Card> CARDS = Cards.getStandardDeck();
+
+
+    /**
+     * Tests that {@link Hand#withAdditionalCard(Card)} maintains the correct score.
+     *
+     * @param handTest contains the test hand, and the target score.
+     */
+    @Theory
+    @SuppressWarnings("checkstyle:methodname")
+    public void addCard_maintainsCorrectScore(HandTestArgument handTest) {
```

Comment:
A test should use the method it tests directly.

Inject a hand and a card as method parameters. Assume that the hand contains no aces, and that the card to add is not an ace.

Call `withAdditionalCard()`.

Test that the resulting hand has the expected difference in points when compared to the original hand.

Also, there is no `addCard` method, so rename this test.